src/context/AppProvider/AppContext.tsx

```diff
@@ -235,7 +247,77 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
     fetchMarketData() // fetch every time assetIds change
     const interval = setInterval(fetchMarketData, 1000 * 60 * 2) // refetch every two minutes
     return () => clearInterval(interval) // clear interval when portfolioAssetIds change
-  }, [dispatch, isPortfolioLoaded, portfolioAssetIds])
+  }, [dispatch, portfolioLoadingState, portfolioAssetIds])
+
+  const toastId = 'accountError'
+
+  const handleAccountErrorToastClose = useCallback(() => {
+    if (toast.isActive(toastId)) {
+      toast.close(toastId)
+    }
+  }, [toast])
+
+  /**
+   * portfolio loading error notification
+   */
+  useEffect(() => {
+    const erroredAccountIds = entries(portfolioLoadingStateGranular).reduce<AccountId[]>(
+      (acc, [accountId, accountState]) => {
+        if (accountState === 'error') acc.push(accountId)
+        return acc
+      },
+      [],
+    )
+
+    if (!erroredAccountIds.length) return // yay
+
+    // we can have multiple accounts with the same name, dont show 'Bitcoin, Bitcoin, Bitcoin'
+    const erroredAccountNames = uniq(
+      erroredAccountIds.map(accountId => assets[accountIdToFeeAssetId(accountId)].name),
+    ).join(', ')
+
+    const handleRetry = () => {
+      handleAccountErrorToastClose()
+      erroredAccountIds.forEach(accountId => {
+        const { chainId, account } = fromAccountId(accountId)
+        const accountSpecifierMap = { [chainId]: account }
+        dispatch(
+          portfolioApi.endpoints.getAccount.initiate(
+            { accountSpecifierMap },
+            { forceRefetch: true },
+          ),
+        )
+      })
+    }
+    const toastOptions = {
+      position: 'top-right' as const,
+      id: toastId,
+      title: translate('common.somethingWentWrong'),
+      status: 'error' as const,
+      description: (
+        <Flex flexDir='column' gap={4} alignItems='flex-start'>
+          <AlertDescription>
+            {translate('common.accountError', { erroredAccountNames })}
+          </AlertDescription>
+          <Button colorScheme='red' onClick={handleRetry}>
+            {translate('common.retry')}
+          </Button>
+        </Flex>
+      ),
+      isClosable: true,
+      duration: null, // don't auto-dismiss
+    }
+    if (!toast.isActive(toastId)) toast(toastOptions)
+    if (toast.isActive(toastId)) return toast.update(toastId, toastOptions)
```

Comment:
if a particular upstream data provider is down, e.g. datahub, or a utxo coinstack, e.g. bitcoin, we can expect multiple accounts to fail at the same time, e.g. cosmos and osmosis, or multiple btc accounts. update the toast rather than toasting a whole loaf of bread.